### ISSUE TITLE
Link for SentencePiece model from streaming convnets recipe

### ISSUE THREAD

**Issue by abhinavkulkarni** [facebookresearch/wav2letter#908]
Hi,

I am looking for the SentencePiece model from [streaming convnets recipe](https://github.com/facebookresearch/wav2letter/tree/v0.2/recipes/models/streaming_convnets/librispeech). The accompanying [README file](https://github.com/facebookresearch/wav2letter/blob/v0.2/recipes/models/streaming_convnets/librispeech/README.md#pre-trained-acoustic-models) seems to have links for tokens and lexicons, but not for the model.

I would like to download the model as I need to create a lexicon (and LM) from my custom dataset.

I tried running the [Python script](https://github.com/facebookresearch/wav2letter/blob/v0.2/recipes/models/utilities/prepare_librispeech_wp_and_official_lexicon.py) to create the WP model, but it yielded slightly different token and lexicon files. Perhaps there's a random element involved in training the WP model and hence my model is (slightly) different from the one that FB team used.

I see that other recipes include links for the WP model in their README, for e.g. [sota/2019 README](https://github.com/facebookresearch/wav2letter/tree/v0.2/recipes/models/sota/2019#tokens-and-lexicon-sets).

Thanks!

**Comment by kerolos** [comment]
Hello @abhinavkulkarni,
in my case i used a WP model from sota/2019 README to create my custom lexicon " https://github.com/facebookresearch/wav2letter/issues/737#issuecomment-654621541 ", and it works without any error. this is because the token files (streaming Convnet and sota/2019 ) are identical. i hope this might help you.

**Comment by abhinavkulkarni** [comment]
Thanks @kerolos!

**Comment by abhinavkulkarni** [comment]
@kerolos: By any chance you

**Comment by abhinavkulkarni** [comment]
@kerolos: By any chance you

**Comment by abhinavkulkarni** [comment]
Hey @kerolos, @tlikhomanenko,

I downloaded the sota/2019 WP model as suggested by @kerolos and the decoder lexicon for beam search from [README](https://github.com/facebookresearch/wav2letter/blob/v0.2/recipes/models/streaming_convnets/librispeech/README.md#pre-trained-acoustic-models) and I notice that the decoder lexicon has different word tokenization what the model would suggest. 

Here's an example for work _"abductions"_:

```bash
$ grep abductions $W2LHOME/decoder/decoder-unigram-10000-nbest10.lexicon
abductions      _ab d uc tion s
abductions      _a b d uc tion s
abductions      _ab d uc t ions
abductions      _ab d u ct ions
abductions      _a b d uc t ions
abductions      _a b d u ct ions
abductions      _ab d u c tion s
abductions      _ab d uc t ion s
abductions      _ab d u ct ion s
abductions      _ab d uc ti ons

$ python
>>> import sentencepiece as spm
>>>
>>> model_name = "/home/w2luser/w2l/am/librispeech-train-all-unigram-10000.model"
>>> sp = spm.SentencePieceProcessor()
>>> sp.Load(model_name)
True
>>>
>>> lm_words = ['abductions']
>>> nbest = 10
>>>
>>> for word in lm_words:
...     wps = sp.NBestEncodeAsPieces(word, nbest)
...     for wp in wps:  # the order matters for our training
...         print(
...             word
...             + "\t"
...             + " ".join([w.replace("\u2581", "_") for w in wp])
...         )
...
abductions      _ab du ction s
abductions      _a b du ction s
abductions      _ab d u ction s
abductions      _a b d u ction s
abductions      _ab d uc tion s
abductions      _a b d uc tion s
abductions      _ab du c tion s
abductions      _ab d uc t ions
abductions      _ ab du ction s
abductions      _a b du c tion s
```

Why does this discrepancy exist? 

Thanks!

**Comment by abhinavkulkarni** [comment]
Hey @kerolos, @tlikhomanenko, 

I am not sure the WP model from sota/2019 can be used for streaming convnets model. 

I get the following tokenization for word "zynools":

```bash
zynools	▁z y no ol s
zynools	▁z yn o ol s
zynools	▁z yn oo l s
zynools	▁z y n o ol s
zynools	▁z y no o l s
zynools	▁z y n oo l s
zynools	▁ z y no ol s
zynools	▁z yn o o l s
zynools	▁ z yn o ol s
zynools	▁ z yn oo l s
```

even though token `yn` is not existent in the sota/2019 lexicon file.

It would be quite helpful if the FB team can provide the WP model used in the streaming convnets recipe.

Thanks!

**Comment by tlikhomanenko** [comment]
hey @abhinavkulkarni,

Sorry for late response (busy with a bunch of things for the flashlight now). Both streaming recipe and sota used the same lexicon and tokens set along with wp model. It should be reproducible if you run exact the same version for sentencepiece + python code on data preparation `python3 ../../utilities/prepare_librispeech_wp_and_official_lexicon.py --data_dst [...] --model_dst [...] --nbest 10 --wp 10000` (here the input data for sentencepiece should be the same and then tokens will be the same, reproduced on several machines before). I have added WP model to sota readme as a lot of people asked for it. And yes tokens and lexicon for sota points to tokens (sorry, I thought I fixed the link already). Let me know if you still cannot get the same lexicon/wp/tokens as in sota/streaming recipes.
